uefi-macros/src/lib.rs

```diff
@@ -181,92 +135,64 @@ pub fn entry(args: TokenStream, input: TokenStream) -> TokenStream {
     let mut f = parse_macro_input!(input as ItemFn);
 
     if let Some(ref abi) = f.sig.abi {
-        errors.append_all(err!(abi, "Entry method must have no ABI modifier"));
+        errors.append_all(err!(abi, "Entry function must have no ABI modifier"));
     }
     if let Some(asyncness) = f.sig.asyncness {
-        errors.append_all(err!(asyncness, "Entry method should not be async"));
+        errors.append_all(err!(asyncness, "Entry function should not be async"));
     }
     if let Some(constness) = f.sig.constness {
-        errors.append_all(err!(constness, "Entry method should not be const"));
+        errors.append_all(err!(constness, "Entry function should not be const"));
     }
     if !f.sig.generics.params.is_empty() {
         errors.append_all(err!(
             f.sig.generics.params,
-            "Entry method should not be generic"
+            "Entry function should not be generic"
         ));
     }
-
-    let signature_span = f.sig.span();
-
-    // If the user doesn't specify any arguments to the entry function, fill in
-    // the image handle and system table arguments automatically.
-    let generated_args = f.sig.inputs.is_empty();
-    if generated_args {
-        f.sig.inputs = parse_quote_spanned!(
-            signature_span=>
-                internal_image_handle: ::uefi::Handle,
-                internal_system_table: *const ::core::ffi::c_void,
-        );
+    if !f.sig.inputs.is_empty() {
+        errors.append_all(err!(f.sig.inputs, "Entry function must have no arguments"));
     }
 
-    let image_handle_ident = get_function_arg_name(&f, 0, &mut errors);
-    let system_table_ident = get_function_arg_name(&f, 1, &mut errors);
-
-    // show most errors at once instead of one by one
+    // Show most errors all at once instead of one by one.
     if !errors.is_empty() {
         return errors.into();
     }
 
-    f.sig.abi = Some(syn::parse2(quote_spanned! (signature_span=> extern "efiapi")).unwrap());
+    let signature_span = f.sig.span();
 
-    // allow the entry function to be unsafe (by moving the keyword around so that it actually works)
-    let unsafety = &f.sig.unsafety;
-    // strip any visibility modifiers
+    // Fill in the image handle and system table arguments automatically.
+    let image_handle_ident = quote!(internal_image_handle);
+    let system_table_ident = quote!(internal_system_table);
+    f.sig.inputs = parse_quote_spanned!(
```

Comment:
nice, thanks for working on this!

(I'm always happy if I do not have to write codegen code by myself.. its tricky 😀)